src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -323,9 +323,10 @@ private bool IsLegacyAdalCacheEnabled(AuthenticationRequestParameters requestPar
             string tenantId,
             InstanceDiscoveryMetadataEntry instanceDiscoveryMetadata)
         {
-            if (IsLegacyAdalCacheEnabled(requestParams))
+            if (msalRefreshTokenCacheItem?.RawClientInfo != null && 
```

Comment:
moved some conditions for being able to save into ADAL legacy cache upfront, to save some ms.